engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -736,267 +736,263 @@ private void execute() throws YarnException, IOException
+      FinalApplicationStatus finalStatus = FinalApplicationStatus.SUCCEEDED;
+      final InetSocketAddress rmAddress = conf.getSocketAddr(YarnConfiguration.RM_ADDRESS,
+          YarnConfiguration.DEFAULT_RM_ADDRESS,
+          YarnConfiguration.DEFAULT_RM_PORT);
 
-    while (!appDone) {
-      loopCounter++;
-      final long currentTimeMillis = System.currentTimeMillis();
+      while (!appDone) {
+        loopCounter++;
+        final long currentTimeMillis = System.currentTimeMillis();
 
-      if (UserGroupInformation.isSecurityEnabled() && currentTimeMillis >= expiryTime && hdfsKeyTabFile != null) {
-        String applicationId = appAttemptID.getApplicationId().toString();
-        expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, principal, hdfsKeyTabFile, credentials, rmAddress, true);
-      }
+        if (UserGroupInformation.isSecurityEnabled() && currentTimeMillis >= expiryTime && hdfsKeyTabFile != null) {
+          String applicationId = appAttemptID.getApplicationId().toString();
+          expiryTime = StramUserLogin.refreshTokens(tokenLifeTime, FileUtils.getTempDirectoryPath(), applicationId, conf, principal, hdfsKeyTabFile, credentials, rmAddress, true);
+        }
 
-      if (currentTimeMillis > nodeReportUpdateTime) {
-        resourceRequestor.updateNodeReports(clientRMService.getNodeReports());
-        nodeReportUpdateTime = currentTimeMillis + UPDATE_NODE_REPORTS_INTERVAL;
-      }
+        if (currentTimeMillis > nodeReportUpdateTime) {
+          resourceRequestor.updateNodeReports(clientRMService.getNodeReports());
+          nodeReportUpdateTime = currentTimeMillis + UPDATE_NODE_REPORTS_INTERVAL;
+        }
 
-      Runnable r;
-      while ((r = this.pendingTasks.poll()) != null) {
-        r.run();
-      }
+        Runnable r;
+        while ((r = this.pendingTasks.poll()) != null) {
+          r.run();
+        }
 
-      // log current state
+        // log current state
       /*
        * LOG.info("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" +
        * numTotalContainers + ", requested=" + numRequestedContainers + ", completed=" + numCompletedContainers +
        * ", failed=" + numFailedContainers + ", currentAllocated=" + this.allAllocatedContainers.size());
        */
-      // Sleep before each loop when asking RM for containers
-      // to avoid flooding RM with spurious requests when it
-      // need not have any available containers
-      try {
-        sleep(1000);
-      } catch (InterruptedException e) {
-        LOG.info("Sleep interrupted " + e.getMessage());
-      }
+        // Sleep before each loop when asking RM for containers
+        // to avoid flooding RM with spurious requests when it
+        // need not have any available containers
+        try {
+          sleep(1000);
+        } catch (InterruptedException e) {
+          LOG.info("Sleep interrupted " + e.getMessage());
```

Comment:
Doesn't look like this is an expected case. Maybe @tweise can shed more light. Will change it to warn and log the trace.